src/main/java/amazin/model/Role.java

```diff
@@ -0,0 +1,39 @@
+package amazin.model;
+
+import javax.persistence.*;
+import java.util.Set;
+
+@Entity
+public class Role {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String name;
+    @ManyToMany(mappedBy = "roles")
+    private Set<User> users;
```

Comment:
How come you put the users in the Role? Rather than the role type in the User?

Reply:
The role has a name and an id and the mapped by I'm just saying that will come from the users